### PR TITLE
replace auto-derived ToJSON instance with one that mirrors ToJSONFields

### PR DESCRIPTION
**Pull request by aymannadeem**

### CODE REVIEW

File: semantic-source/src/Source/Range.hs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE DeriveAnyClass, DeriveGeneric, RankNTypes #-}
+{-# LANGUAGE DeriveAnyClass, DeriveGeneric, RankNTypes, NamedFieldPuns, OverloadedStrings #-}
```
Comment (robrix) [inline semantic-source/src/Source/Range.hs]: Is `OverloadedStrings` necessary here?

File: semantic-source/src/Source/Loc.hs [hunk 2]
```diff
@@ -13,6 +13,7 @@ import GHC.Generics (Generic)
 import Prelude hiding (span)
 import Source.Range
 import Source.Span
+import Data.Aeson (ToJSON(..), object, (.=))
```
Comment (robrix) [inline semantic-source/src/Source/Loc.hs]: Could you please alphabetize the imports? 🙏

Comment (aymannadeem) [reply semantic-source/src/Source/Loc.hs]: yep my bad!

File: semantic-source/src/Source/Range.hs (hunk 1 above)
Comment (aymannadeem) [reply semantic-source/src/Source/Range.hs]: Removed.

**Pull request merged**